### PR TITLE
Emit event when running ./occ db:add-missing-indices

### PR DESCRIPTION
**Pull request by rullzer**
This allows apps to listen to this event in order to also update
indecies there.

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: core/Command/Db/AddMissingIndices.php [hunk 1]
```diff
@@ -41,10 +44,13 @@ class AddMissingIndices extends Command {
...
 	private $connection;
 
+	/** @var EventDispatcherInterface */
+	private $dispatcher;
+
 	/**
 	 * @param IDBConnection $connection
 	 */
-	public function __construct(IDBConnection $connection) {
+	public function __construct(IDBConnection $connection, EventDispatcherInterface $dispatcher) {
```
Comment (juliushaertl) [inline core/Command/Db/AddMissingIndices.php]: You're missing the assignment of `$dispatcher` to the private field.

**Review by juliusknorr — CHANGES_REQUESTED** [review]

File: core/Command/Db/AddMissingIndices.php (hunk 1 above)
Comment (rullzer) [reply core/Command/Db/AddMissingIndices.php]: *look again*

**Review by juliusknorr — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]
Works 👍

File: lib/public/IDBConnection.php [hunk 2]
```diff
@@ -45,6 +45,9 @@
  * @since 6.0.0
  */
 interface IDBConnection {
+
+	const ADD_MISSING_INDEXES_EVENT = this::class . '::ADD_MISSING_INDEXES';
```
Comment (MorrisJobke) [inline lib/public/IDBConnection.php]:
```
lib/public/IDBConnection.php:49 PhanUndeclaredClassConstant Reference to constant class from undeclared class \OCP\this
```

Comment (MorrisJobke) [reply lib/public/IDBConnection.php]: I fixed it by replacing the `this` with `self` 😉

**Comment by rullzer** [comment]
Would a backport make sense? it is a small change but then apps can already add indexes for 13 apps.

**Comment by MorrisJobke** [comment]
> Would a backport make sense? it is a small change but then apps can already add indexes for 13 apps.

I would say so.

**Comment by rullzer** [comment]
backport in https://github.com/nextcloud/server/pull/9620

**Pull request merged**